serialization/serialization.go

```diff
@@ -0,0 +1,291 @@
+// Package serialization is for working with the mysql::serialization format
+//
+// mysql::serialization is a serialization format introduced with tagged GTIDs
+//
+// https://dev.mysql.com/doc/dev/mysql-server/latest/PageLibsMysqlSerialization.html
+package serialization
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"slices"
+	"strings"
+)
+
+// Message is a mysql::serialization message
+type Message struct {
+	Version uint8 // >= 0
+	Format  Format
+}
+
+func (m *Message) String() (text string) {
+	text += fmt.Sprintf("Message (version: %d)", m.Version)
+	for _, line := range strings.Split(m.Format.String(), "\n") {
+		text += "\n  " + line
+	}
+	return
+}
+
+// GetFieldByName returns a field if the name matches and an error if there is no match
+func (m *Message) GetFieldByName(name string) (Field, error) {
+	for _, f := range m.Format.Fields {
+		if f.Name == name {
+			return f, nil
+		}
+	}
+	return Field{}, fmt.Errorf("field not found: %s", name)
+}
+
+// Format is describing a `message_format`
+type Format struct {
+	Size                  uint64
+	LastNonIgnorableField int
+	Fields                []Field
+}
+
+func (f *Format) String() (text string) {
+	text += fmt.Sprintf("Format (Size: %d, LastNonIgnorableField: %d)\n",
+		f.Size, f.LastNonIgnorableField)
+	for _, f := range f.Fields {
+		text += fmt.Sprintf("Field %02d (Name: %s, Skipped: %t, Type: %T)\n",
+			f.ID, f.Name, f.Skipped, f.Type)
+		if f.Type != nil {
+			text += fmt.Sprintf("  Value: %s\n", f.Type.String())
+		}
+	}
+	return text
+}
+
+type Field struct {
+	ID       int
+	Type     FieldType
+	Optional bool
+	Name     string
+	Skipped  bool
+}
+
+type FieldType interface {
+	fmt.Stringer
+}
+
+type FieldIntFixed struct {
```

Comment:
this type is called "...Fixed", I guess it means it's fixed size encoded? But it's implemented as a variable-length encoded.